### ISSUE TITLE
frame rate and duration of videos

### ISSUE THREAD

**Issue by amirsafaei** [DAMO-NLP-SG/Video-LLaMA#27]
Based on my understanding, you use only 8 frames sampled uniformly across the video. I have the following questions :
1. does increasing the total frames by a lot slow down the training significantly ? my own conclusion was that it shouldn't because the features are extracted frame by frame and then concatenated. so it won't add to learnable parameters. is this correct ?
2. what if I want to fix the frame rate instead of the total frames? so that I can account for longer or shorter videos but maintain a certain frame rate on them. is this possible with your model?
3. if 2 is possible in theory, do you think your arch would work well on videos that need a higher sampling rate?
4. does your model treat videos as 8 separate images and process them one by one or does it take into account the whole 8 frames at once?
5. if I want to use other models for feature extraction of videos, can I still use your arch ? like instead of using raw videos using features that are already extracted from frozen models. (I think in a way this is what you do too by using blip-2)
6. last but not least, where does learning actually happen in your work? there are a lot of pre-trained and frozen models in use, I'm kinda confused about what part gets updated based on fine-tuning.

I hope you could help me understand your work better and hopefully be able to use it on my own data.

**Comment by SushantGautam** [comment]
Also, I think there should be a config param. to specify total frames. 
We could measure the performance differences too.

**Comment by hangzhang-nlp** [comment]
1. Increasing the total number of frames will indeed increase the training time. Although it won't add to learnable parameters, the amount of computation involved in encoding video will increase.
2. You can try to fix the frame rate. This modification is relatively easy to make in our framework. Since the video lengths in the training set vary significantly, some truncation and padding operations will be required.
3.  Using a higher frame length to pre-train Video-LLaMA may be better, but the training time will be longer. In some video pre-training works, such as the one mentioned in Figure 3 of the CLIP4Clip paper, it has been observed that increasing the frame length beyond eight frames only yields a marginal improvement.
4. In the visual encoder (ViT and Q-Former), the frames are encoded independently. However, in the video Q-Former, the features of these frames are fused, resulting in an overall representation of the video.
6.  Sure, you can use pre-extracted features from other models to replace the BLIP-2 module in our architecture.
7.  The learnable modules in the Vision-Language Branch include: a) the position embedding layer, b) the Video Q-Former, and c) the Linear layer. These modules are optimized to effectively connect the output of the frozen visual encoder (BLIP-2) to the frozen LLMs (Language Model Modules)

References: Clip4Clip: https://arxiv.org/abs/2104.08860
